redo.dev/src/components/demos/stackblitzGenerators/stackblitzIndexFileBuilder.ts

```diff
@@ -0,0 +1,28 @@
+export const stackblitzIndexFileBuilder = (embedId: string) => {
+    const demoInfo = demoAdapters[embedId]
+    const fileImports = demoInfo[0]
+    const dataObject = demoInfo[1]
+
+    return `import {populateDemo} from "./populateDemo"
+${fileImports}
+populateDemo(${dataObject})`
+}
+
+const demoAdapters: Record<string, string[]> = {
```

Comment:
`string[]` could be `[]` or have 1000 members. In the case where the list needs to have a specific number of items, you can use a tuple type, which also allows you to name each item to help clarify meaning:

```ts
// We can share this type across all files that need it
export type EmbedId = "model" | "space" | "declaration" | "constraints"

type DemoAdapter = [
    importFromDemo: string,
    dataFromImports: string    
]

const demoAdapters: Record<EmbedId, DemoAdapter> = {
    ...
}
```

Whenever you can a more specific type like this, it means we have to write less tests and worry about less edge cases. When that's not possible, it's best to give a very generic type like `Record<string, string[]>` an explicit name like `DemoAdapters` to help clarify its meaning. If it's only used in one place alongside a variable of the same name (e.g. `demoAdapters` and `DemoAdapters`), it's best to provide no explicit type and just let TS infer the type automatically, which will be more accurate since it will still tell you which keys exist, whereas `Record<string, string[]>` will allow you to access any string as a property.

This is actually a very minor detail, but it just comes up so much I figured I'd take the chance to write an essay 🤣